### DISCUSSION TITLE
"Could not connect to Portainer" Error

### DISCUSSION THREAD

**Discussion by MikeDeltaHH** [linuxserver/Heimdall#769] [App issues]
Hello!

I have Heimdall and Portainer running on the same Docker host and added Portainer as an application! I can then also access Portainer via the link and have stored the login data for a live display in the config of the application - however, I only get the error message "Could not connect to Portainer" when testing - I have already tried everything possible and do not get it...

The Portainer URL:
https://192.168.2.126:9443

Any idea?

Regards!
Mike

**Comment by j0nnymoe** [comment]
There could be 2 issues here:

1) something within portainer has changed with the API which means the App within Heimdall needs to be fixed
2) you've got Heimdall and/or portainer on dedicated IP's via macvlan which blocks the communication

**Comment by MikeDeltaHH** [comment]
Hello!

Thanks for your tips...

Portainer & Heimdall are running in bridge mode - I have another Portainer container running on another Docker host with the same Portainer version and mounted as an application, there is not this problem - the only difference between the two Portainer instances is that the other instance is reachable via port 9000 and via HTTP:

Does not work:
https://192.168.2.126:9443

Works:
http://192.168.2.125:9000

I can't customize the portainer instance, I would have to shutdown Portainer completely which is too much effort for me...

**Comment by Typ1er** [comment]
Same problem here, password check does not work

**Comment by gerardsyd** [comment]
Any luck with this / updates? thanks!

**Comment by ISeeTWizard** [comment]
I had the same and after restarting my Portainer instances problem was solved
I don't know why but it reworks now...

**Comment by jovington** [comment]
Hello !!
I have been trying to connect portainer to Heimdall and it tells me there are credential issues.
I have two ports

9443 https
8000 http

the connection url is http://xxx.xxx.xxx.xx:8000 and I put the username and password, I do the test and it says "Invalid credentials"

Any solution ?

Thanks in advance !!

**Comment by keriati** [comment]
Hello!

I am actively using the Portainer integration and it works without issues.
I created a read only user for Portainer for this and use that user.
Also for me Portainer runs on port 9443.

If you can, check your Laravel logs in `storage/logs` (or if you are using docker version in `config/log/heimdall` on your data volume) and submit here. 

Also check in your browser in the network tab the response from the the `test_config` request and if there is some more detailed error, submit it here.

To get more detailed errors I also recommend setting `APP_DEBUG=true` in the .env file.

**Reply by gerardsyd** [reply]
Yep managed to get mine to work as well. Don't recall what I did - think it had to do with using the web address vs container name
